test/h5test.c

```diff
@@ -2193,12 +2435,38 @@ h5_using_native_vol(hid_t fapl_id, hid_t obj_id, bool *is_native_vol)
     return ret_value;
 }
 
+/*-------------------------------------------------------------------------
+ * Function:    h5_get_test_driver_name
+ *
+ * Purpose:     Checks the HDF5_DRIVER and HDF5_TEST_DRIVER environment
+ *              variables to see if a driver name has been set for testing.
+ *
+ * Return:      Driver name if set/NULL otherwise
+ *
+ *-------------------------------------------------------------------------
+ */
+const char *
+h5_get_test_driver_name(void)
+{
+    char *envval;
+
+    assert(H5_DEFAULT_VFD == H5FD_SEC2);
+
+    if ((envval = getenv(HDF5_DRIVER)))
```

Comment:
This seems to be mixing the HDF5_DRIVER source macro and the environment variable "HDF5_DRIVER".

Reply:
I agree - I think the HDF5_DRIVER macro should be removed.  That would clean up the code in this area quite a bit.

Reply:
If it's your new macro, you should have one for the HDF5_TEST_DRIVER environment variable also.

Reply:
I would also suggest putting 'ENV' somewhere in their names, so that it's a little more clear what their purpose is.

Reply:
I'm not sure I understand. As far as I remember, `HDF5_DRIVER` never existed anywhere in the library or configure code until it was added as `HDF5_DRIVER` = "HDF5_DRIVER". One could have passed it as a compile option, but I don't think it was documented and we definitely didn't have a test for that at the time I worked on this previously. So it seems there's not much to remove here other than the one `#ifdef HDF5_DRIVER` I removed in this PR. `HDF5_DRIVER` is also in H5public.h so I'd prefer not to change it, but open to any discussion there.

No problem adding a macro for "HDF5_TEST_DRIVER", though my intention for this is for it to be a temporary environment variable, so that one would probably go in a private header somewhere. Using "HDF5_TEST_DRIVER" is just a way for CMake and Autotools to play nicely together since Autotools doesn't pick and choose which tests to run with all the VFDs and `HDF5_DRIVER` works similar to how `HDF5_VOL_CONNECTOR` does for setting something on a FAPL. Because of that, `make check-vfd` causes Autotools to run every single test with all the VFDs and fails in a lot of places while CMake is conservative about what it tests. It was an oversight at the time I implemented the VFD plugins feature.

Reply:
I should clarify that by temporary, I mean that in the future we need to refactor what tests get run for VFD testing so Autotools can test more things. But until then, "HDF5_TEST_DRIVER" is a stopgap that lets CMake keep generally testing the same things it was with VFDs, while preventing Autotools from running too many tests against VFDs.